extension/src/popup/components/sendPayment/SendConfirm/SubmitResult/index.tsx

```diff
@@ -120,121 +114,178 @@ export const SubmitSuccess = ({ viewDetails }: { viewDetails: () => void }) => {
 };
 
 export const SubmitFail = () => {
-  const {
-    destinationBalances,
-    error,
-    transactionData: {
-      destination,
-      federationAddress,
-      amount,
-      asset,
-      destinationAsset,
-    },
-    assetIcons,
-  } = useSelector(transactionSubmissionSelector);
-  const isPathPayment = useSelector(isPathPaymentSelector);
+  const { error } = useSelector(transactionSubmissionSelector);
   const isSwap = useIsSwap();
   const { t } = useTranslation();
 
-  const sourceAsset = getAssetFromCanonical(asset);
-
   useEffect(() => {
     emitMetric(METRIC_NAMES.sendPaymentError, { error });
   }, [error]);
 
-  const decideError = () => {
-    // unfunded destination
-    if (
-      !isSwap &&
-      shouldAccountDoesntExistWarning(
-        destinationBalances.isFunded || false,
-        asset,
-        amount,
-      )
-    ) {
-      return <AccountDoesntExistWarning />;
+  const getErrorDetails = (err: ErrorMessage | undefined) => {
+    const errorDetails = {
+      title: "",
+      errorBlock: <div></div>,
+      opError: RESULT_CODES.tx_failed,
+      status: "",
+    };
```

Comment:
might be nice to create an interface so it's really clear what the type of `errorDetails` is

Reply:
👍  https://github.com/stellar/freighter/pull/584/commits/3ad2b4a55bef244a93070b0772606b9a7e746775